api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -160,19 +160,29 @@ private String createFirecloudUserAndBillingProject(User user) {
     return createFirecloudBillingProject(user);
   }
 
-  private User saveUserWithConflictHandling(User user) {
-    if (user.getGivenName().length() > 80) {
-      throw new BadRequestException("Given Name length exceeds character limit. (80)");
-    }
-    if (user.getFamilyName().length() > 80) {
-      throw new BadRequestException("Family Name length exceeds character limit. (80)");
+  private void validateStringLength(String field, String fieldName, int max, int min) {
+    if (field == null) {
+      throw new BadRequestException(String.format("%s cannot be left blank!", fieldName));
     }
-    if (user.getCurrentPosition() != null && user.getCurrentPosition().length() > 255) {
-      throw new BadRequestException("Current Position length exceeds character limit. (255)");
+    if (field.length() > max) {
+      throw new BadRequestException(String.format("%s length exceeds character limit. (%d)", fieldName, max));
     }
-    if (user.getOrganization() != null && user.getOrganization().length() > 255) {
-      throw new BadRequestException("Organization length exceeds character limit. (255)");
+    if (field.length() < min) {
+      if (min == 1) {
```

Comment:
Shouldn't this be 0?

Reply:
no bc 1 character is the minimum character length.  we don't want to allow blank fields (0 characters)

Reply:
OK, I wasn't following the logic initially. This does seem pretty round-about though; could you instead just check `field.length() == 0` ?

Reply:
true.  i wanted it to be robust to handle different mins in case we decided to implement them down the road, but i thought it would be nice to have a custom message for the empty case.  i can get rid of that though - what do you think?

Reply:
I can't really imagine cases other than the two you have here, but I don't really feel strongly about this, I'll leave it up to your judgement.